pkg/runtime/runtime.go

```diff
@@ -1521,12 +1521,15 @@ func (a *DaprRuntime) publishMessageHTTP(ctx context.Context, msg *pubsubSubscri
 	req.WithRawData(msg.data, contenttype.CloudEventContentType)
 	req.WithCustomHTTPMetadata(msg.metadata)
 
-	if cloudEvent[pubsub.TraceIDField] != nil {
-		traceID := cloudEvent[pubsub.TraceIDField].(string)
-		sc, _ := diag.SpanContextFromW3CString(traceID)
-		spanName := fmt.Sprintf("pubsub/%s", msg.topic)
-		ctx, span = diag.StartInternalCallbackSpan(ctx, spanName, sc, a.globalConfig.Spec.TracingSpec)
+	traceParent, _ := cloudEvent[pubsub.TraceParentField].(string)
```

Comment:
In order to preserve backwards compat, we need to check for either trace parent or trace ID.

Reply:
This method name is a bit confusing - this is Dapr sending the message to the consumer.